lib/src/package_analysis/common.dart

```diff
@@ -1,12 +1,235 @@
-import 'package:analyzer/dart/analysis/analysis_context.dart';
-import 'package:analyzer/dart/analysis/analysis_context_collection.dart';
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:analyzer/dart/analysis/session.dart';
+import 'package:analyzer/file_system/file_system.dart' as resource;
+import 'package:pana/pana.dart';
+import 'package:path/path.dart' as path;
+import 'package:pub_semver/pub_semver.dart';
+import 'package:pubspec_parse/pubspec_parse.dart' hide Pubspec;
+
+const indentedEncoder = JsonEncoder.withIndent('  ');
 
 abstract class PackageAnalysisContext {
-  AnalysisContextCollection get analysisContextCollection;
+  AnalysisSession get analysisSession;
+
+  /// The name of the package being analyzed.
+  late final String packageName = pubspec.name;
+
+  /// The name of the singular dependency of this target dummy package, or null
+  /// if the package being analysed does not use a dummy.
+  final String? targetPackageName;
+
+  /// The absolute path of the package being analyzed.
+  final String packagePath;
 
-  AnalysisContext contextFor(String path) =>
-      analysisContextCollection.contextFor(path);
+  /// The absolute path of the target package, given that the package being
+  /// analysed uses a dummy.
+  late final String? targetPackagePath = getDependencyDirectory(this, targetPackageName!);
 
-  /// Log [message] as warning that something unexpected happened.
+  /// The pubspec of the package being analyzed.
+  late final Pubspec pubspec =
+      Pubspec.parseYaml(readFile(path.join(packagePath, 'pubspec.yaml')));
+
+  /// A [Map] storing metadata about the hosted dependencies of the package
+  /// being analyzed.
+  late final Map<String, HostedDependency> dependencies = Map.fromEntries(
+      pubspec
+          .dependencies.entries
+          .where((dependency) => dependency.value is HostedDependency)
+          .map(
+              (entry) => MapEntry(entry.key, entry.value as HostedDependency)));
+
+  /// A [Map] storing metadata about the hosted dependencies of the package
+  /// being analyzed, given that the package associated with this
+  /// [PackageAnalysisContext] is a dummy package with a single non-transitive
+  /// dependency.
+  late final Map<String, HostedDependency> targetDependencies = Map.fromEntries(
```

Comment:
Why do we need both?

Reply:
addressed in https://github.com/CicadaCinema/pana/pull/2/commits/7e09995c47283da2c271325567dc918abe7f00c8